# CSC 418 Group D Data Science Project

Category : **Major**
Group Members
 
1. P15/5620/2019 : Njagi Baraka Fadhili
2. P15/1636/2019 : Kabiru Sharleen Njeri
3. P15/1635/2019 : Obora Melanie Fayne
4. P15/137631/2019 : Ali Amina Abdi
5. P15/130607/2018 : Munyao Mary June

## Importing libraries
_____

In [5]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import f1_score,accuracy_score,classification_report,confusion_matrix ,recall_score, precision_score, roc_curve, roc_auc_score

## Reading Files

____

In [7]:
train_data = pd.read_csv("./data/train.csv")
# Preview the first five rows of the train dataset
print(f'The shape of the dataset is: {train_data.shape}')
train_data.head()

The shape of the dataset is: (4459, 4993)


ID      target  48df886f9  0deb4b6a8  34b15f335  a8cb14b00  \
0  000d6aaf2  38000000.0        0.0          0        0.0          0   
1  000fbd867    600000.0        0.0          0        0.0          0   
2  0027d6b71  10000000.0        0.0          0        0.0          0   
3  0028cbf45   2000000.0        0.0          0        0.0          0   
4  002a68644  14400000.0        0.0          0        0.0          0   

   2f0771a37  30347e683  d08d1fbe3  6ee66e115  ...  3ecc09859  9281abeea  \
0          0          0          0          0  ...        0.0        0.0   
1          0          0          0          0  ...        0.0        0.0   
2          0          0          0          0  ...        0.0        0.0   
3          0          0          0          0  ...        0.0        0.0   
4          0          0          0          0  ...        0.0        0.0   

   8675bec0b  3a13ed79a  f677d4d13  71b203550  137efaa80  fb36b89d9  \
0        0.0          0          0          0          0          0   
1        0.0          0          0          0          0          0   
2        0.0          0          0          0          0          0   
3        0.0          0          0          0          0          0   
4        0.0          0          0          0          0          0   

   7e293fbaf  9fc776466  
0          0          0  
1          0          0  
2          0          0  
3          0          0  
4          0          0  

[5 rows x 4993 columns]

In [8]:
test_data = pd.read_csv("./data/test.csv")
# Preview the first five rows of the test dataset
print(f'The shape of the dataset is: {test_data.shape}')
test_data.head()

The shape of the dataset is: (49342, 4992)


ID  48df886f9  0deb4b6a8  34b15f335  a8cb14b00  2f0771a37  \
0  000137c73        0.0        0.0        0.0        0.0        0.0   
1  00021489f        0.0        0.0        0.0        0.0        0.0   
2  0004d7953        0.0        0.0        0.0        0.0        0.0   
3  00056a333        0.0        0.0        0.0        0.0        0.0   
4  00056d8eb        0.0        0.0        0.0        0.0        0.0   

   30347e683  d08d1fbe3  6ee66e115  20aa07010  ...  3ecc09859  9281abeea  \
0        0.0        0.0        0.0        0.0  ...        0.0        0.0   
1        0.0        0.0        0.0        0.0  ...        0.0        0.0   
2        0.0        0.0        0.0        0.0  ...        0.0        0.0   
3        0.0        0.0        0.0        0.0  ...        0.0        0.0   
4        0.0        0.0        0.0        0.0  ...        0.0        0.0   

   8675bec0b  3a13ed79a  f677d4d13  71b203550  137efaa80  fb36b89d9  \
0        0.0        0.0        0.0        0.0        0.0        0.0   
1        0.0        0.0        0.0        0.0        0.0        0.0   
2        0.0        0.0        0.0        0.0        0.0        0.0   
3        0.0        0.0        0.0        0.0        0.0        0.0   
4        0.0        0.0        0.0        0.0        0.0        0.0   

   7e293fbaf  9fc776466  
0        0.0        0.0  
1        0.0        0.0  
2        0.0        0.0  
3        0.0        0.0  
4        0.0        0.0  

[5 rows x 4992 columns]

 *   We are provided with an anonymized dataset containing numeric feature variables, the numeric target column, and a string ID column

* The train data and test data  has 4992 unique Columns 
* the train data has 4459 rows 
* the test data has 49342 rows 
* In the Train data , the Number of columns is more than the number of train rows.
* Test data is almost 10 times as that of train set.

### Splitting the train dataset for training

In [ ]:
# Select main columns to be used in training
main_cols = train_data.columns.difference(['ID_code', 'target'])
X = train_data[main_cols]
y = train_data.target

In [ ]:
# Split data into train and test sets
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.3, random_state=42)

print(X_train.shape)
print(X_test.shape)

## Baseline Model using Logistic Regression

In [ ]:
model_lr = LogisticRegression()
model_lr.fit(X_train, y_train)

In [ ]:
# Prediction
y_pred_lr = model_lr.predict(X_test)

In [ ]:
print('Accuracy is : ', accuracy_score(y_test, y_pred_lr))
print(f'F1 score on the X_test is: {f1_score(y_test, y_pred_lr)}')
print(' recall:', recall_score(y_test, y_pred_lr))
print(' precision:',precision_score(y_test, y_pred_lr))
print('Area under the ROC curve:' , roc_auc_score(y_test, y_pred_lr))
confusion = confusion_matrix(y_test, y_pred_lr)
print(f'Confusion Matrix on the X_test is:\n {confusion}')

### Predicting The Test Dataset

In [ ]:
# Make prediction on the test set
test_df = test_data[main_cols]
predictions = model_lr.predict(test_df)

In [ ]:

sample_sub = pd.read_csv('/data/sample_submission.csv')
# Create a submission file
sub_file = sample_sub.copy()
sub_file.predictions = predictions

In [ ]:
# Check the distribution of our predictions
sns.countplot(sub_file.predictions)

In [ ]:
sub_file.to_csv('Unprocessed_Lr_Submission.csv', index = False)